backend/mobilityAI/Controllers/PatientsController.cs

```diff
@@ -163,6 +170,71 @@ public IActionResult GetPatientActivity(int patientId, long start, long end)
             return Ok();
         }
 
+        /// <summary>
+        /// Adding patient achievements. If current achievement stats exist, update the information. 
+        /// If it doesn't exist, add a new row to the database for that patient
+        /// </summary>
+        /// <param name="patientId">
+        /// The id value of the patient
+        /// </param>
+        /// <param name="steps">
+        /// The number of steps goal that the patient wishes to achieve 
+        /// </param>
+        /// <param name="activityTime">
+        /// The number of steps goal that the patient wishes to achieve
+        /// </param>
+        /// <returns>
+        /// 200 if successful
+        /// </returns>
+        [HttpPost("PatientAchievements")]
+        public IActionResult PatientAchievements(int patientId, int steps, int activityTime)
+        {
+            Achievement dataQuery = (from a in _context.Achievements
+                                 where (a.Id == patientId)
+                                 select a).SingleOrDefault();
+
+            if (dataQuery == null)
+            {
+                Achievement data = new Achievement();
+
+                data.Id = patientId;
+                data.Steps = steps;
+                data.ActivityTime = activityTime;
+
+                _context.Achievements.Add(data);
+            } else {
+                dataQuery.Id = patientId;
+                dataQuery.Steps = steps;
+                dataQuery.ActivityTime = activityTime;
+            }
+
+            _context.SaveChanges();
+            return Ok();
+        }
+
+        /// <summary>
+        /// Returning the patient achievements stats
+        /// </summary>
+        /// <param name="patientId">
+        /// The id value of the patient
+        /// </param>
+        /// <returns>
+        /// If the patient does not have any achievement stats, returns -1
+        /// If the patient has achievement stats, returns the stats
+        /// </returns>
+        [HttpGet("GetPatientAchievements")]
+        public JsonResult GetPatientAchievements(int patientId) {
+            Achievement data = (from a in _context.Achievements
+                       where a.Id == patientId
+                       select a).SingleOrDefault();
+
+            if (data == null) {
+                return new JsonResult(-1);
```

Comment:
We can return a more descriptive error result like "Could not find any achievement data"